### PR TITLE
Replace OrderByList by std::vector<OrderBy> and Skip nested structure inside IN filter test case

### PR DESCRIPTION
**Pull request by cherylEnkidu**
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: firestore/integration_test_internal/src/query_test.cc [hunk 1]
```diff
@@ -660,6 +660,7 @@ TEST_F(QueryTest, TestQueriesCanUseArrayContainsFilters) {
 }
 
 TEST_F(QueryTest, TestQueriesCanUseInFilters) {
+  GTEST_SKIP() << "Skipping this test until double free bug is fixed.";
```
Comment (DellaBitta) [inline firestore/integration_test_internal/src/query_test.cc:663]: Could you add the b/# here so we can track it?

Comment (dconeybe) [reply firestore/integration_test_internal/src/query_test.cc:663]: +1. Also, please shorten "Skipping this test" to just "Skip".

Comment (cherylEnkidu) [reply firestore/integration_test_internal/src/query_test.cc]: done

**Comment by DellaBitta** [comment]
Looks like some Database and Auth tests are failing on Windows, but we can ignore those errors since this PR doesn't affect those products.

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**